src/Access/HTTPAuthClient.h

```diff
@@ -62,12 +64,14 @@ class HTTPAuthClient
     }
 
     const Poco::URI & getURI() const { return uri; }
+    const std::vector<String> & getHeaders() const { return headers; }
 
 private:
     const ConnectionTimeouts timeouts;
     const size_t max_tries;
     const size_t retry_initial_backoff_ms;
     const size_t retry_max_backoff_ms;
+    const std::vector<String> headers;
```

Comment:
Let's please rename `headers`/`getHeaders` to `forward_headers`/`getForwardHeaders`, otherwise it's a bit difficult to understand in the `authenticate` which headers represent what exactly